src/wallet/coincontrol.h

```diff
@@ -19,12 +21,12 @@ class CCoinControl
     bool fAllowOtherInputs;
     //! Includes watch only addresses which match the ISMINE_WATCH_SOLVABLE criteria
     bool fAllowWatchOnly;
-    //! Override estimated feerate
+    //! Override automatically calculated feerate, nFeeRate must be set if true
     bool fOverrideFeeRate;
-    //! Feerate to use if overrideFeeRate is true
-    CFeeRate nFeeRate;
-    //! Override the default confirmation target, 0 = use default
-    int nConfirmTarget;
+    //! Override the default payTxFee if set, 0 = use smart fee estimation
+    boost::optional<CFeeRate> m_feerate;
+    //! Override the default confirmation target
+    boost::optional<unsigned int> m_confirm_target;
```

Comment:
I hate the magic values of 0.  Actually I think I should make the logic even slightly more clear.
There should just be a clear parameter precedence.
- coin_control.m_feerate
- coin_control.m_confirm_target
- ::payTxFee (global)
- ::nTxConfirmTarget (globa)

It goes down the list until it fines one that is set and all the first 3 should be boost::optional's.  Why do you not like boost::optional?
I could skip changing the global payTxFee for now, and still have that work via the 0 magic value.
The code to process this will be slightly longer in `GetMininumFee` but I think it'll be far more clear no?